File: ollama/codigo-final/mandelbrot/qwen2.5-coder-7b/cot/temp_0.7/mandelbrot.py
import sys
import struct

# Function to calculate if a point belongs to the Mandelbrot set
def mandelbrot(c, max_iter):
    zr, zi = 0.0, 0.0
    lim_sq = 2.0 * 2.0
    for n in range(max_iter):
        zr_new = zr*zr - zi*zi + c[0]
        zi_new = 2*zr*zi + c[1]
        if zr_new*zr_new + zi_new*zi_new > lim_sq:
            return n
        zr, zi = zr_new, zi_new
    return max_iter

# Main function to generate Mandelbrot set and output PBM P4 format
def main():
    N = int(sys.argv[1])
    max_iter = 50
    lim_sq = 2.0 * 2.0
    
    # Generate the bitmap data
    bitmap = bytearray()
    for y in range(N):
        for x in range(N):
            c = (2.0 * x / N - 1.5, 2.0 * y / N - 1.0)
            iter_count = mandelbrot(c, max_iter)
            pixel_value = 0 if iter_count == max_iter else 1
            bitmap.append(pixel_value)
        
        # Pad the last byte of each row with zeros on the right if needed
        if N % 8 != 0:
            padding = 8 - (N % 8)
            bitmap.extend([0] * padding)

    # Output PBM P4 header and binary pixel data to stdout
    sys.stdout.write(f"P4\n{N} {N}\n")
    sys.stdout.buffer.write(bitmap)

if __name__ == "__main__":
    main()

File: ollama/codigo-final/mandelbrot/qwen2.5-coder-7b/cot/temp_0.7/mandelbrot.java
import java.io.*;

public class Mandelbrot {
    public static void main(String[] args) throws IOException {
        if (args.length != 1) {
            System.out.println("Usage: java Mandelbrot N");
            return;
        }

        int N = Integer.parseInt(args[0]);
        double limSquared = 4.0; // Escape threshold squared
        int maxIterations = 50;

        // Output PBM header
        System.out.println("P4\n" + N + " " + N);

        for (int y = 0; y < N; y++) {
            for (int x = 0; x < N; x++) {
                double zr = 0, zi = 0;
                double cr = 2.0 * x / N - 1.5;
                double ci = 2.0 * y / N - 1.0;

                int iter;
                for (iter = 0; iter < maxIterations && (zr*zr + zi*zi <= limSquared); iter++) {
                    double zrNew = zr*zr - zi*zi + cr;
                    zi = 2.0 * zr * zi + ci;
                    zr = zrNew;
                }

                // Write the pixel data
                if (iter == maxIterations) {
                    System.out.print((char)0); // Bit set, pixel belongs to the set
                } else {
                    System.out.print((char)128); // Bit cleared, pixel not in the set
                }
            }

            // Pad with zeros if N is not divisible by 8
            for (int i = 0; i < (N % 8); i++) {
                System.out.print((char)0);
            }
        }
    }
}
